rest-client/src/main/java/org/sagebionetworks/bridge/rest/AuthenticationHandler.java

```diff
@@ -54,7 +56,11 @@ public AuthenticationHandler(SignIn signIn, UserSessionInfoProvider userSessionI
         }
         tryCount++;
 
-        userSession = userSessionInfoProvider.retrieveSession(signIn);
+        try {
+            userSession = userSessionInfoProvider.retrieveSession(signIn);
+        } catch (ConsentRequiredException e) {
+            LOG.debug("Ignoring 412 during auto-authentication");
```

Comment:
I thought this might be the case also, until I saw what would happen to this line:
`user.getClient(ParticipantsApi.class).getUsersParticipantRecord().execute();`

If the user's session is current, the call succeeds.

If the user's session is invalid (expired, or logged out), the call would throw a 412, even though the API being called does not require consent. The call would fail due to something that we're aiming to encapsulate (auto-reauth in case of session expiry), so throwing a 412 sort of an unexpected side effect. Running the exact same line again would succeed.

I'm leaning towards hiding this exception. Unless there is a use case where we'd want to know the participant has not consented, when we're calling an API that doesn't require consent (if it did, a 412 would get thrown anyway, and the stack trace would show it coming from the API being called, instead of an authenticator being called internally by OkHttp)